sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/arithmetic.scala

```diff
@@ -592,7 +599,10 @@ case class Multiply(
   override def inputType: AbstractDataType = NumericType
 
   override def symbol: String = "*"
-  override def decimalMethod: String = "$times"
+
+  override def decimalMethod(mathContextValue: GlobalValue, value1: String, value2: String):
+  String = s"Decimal.apply($value1.toJavaBigDecimal()" +
+    s".multiply($value2.toJavaBigDecimal(), $mathContextValue))"
 
```

Comment:
Doesn't the non-codegen path also need to be updated? It runs a numeric.times which is implemented as

https://github.com/apache/spark/blob/46332d985e52f76887c139f67d1471b82e85d5ca/sql/catalyst/src/main/scala/org/apache/spark/sql/types/Decimal.scala#L495-L496

where the `MATH_CONTEXT` is still the wrong one.

https://github.com/apache/spark/blob/46332d985e52f76887c139f67d1471b82e85d5ca/sql/catalyst/src/main/scala/org/apache/spark/sql/types/Decimal.scala#L548

Reply:
Thanks for remind, I add test for non-codegen and change the logic for `nullSafeEval` to adapt non-codegen. 
Can't change the logic in https://github.com/apache/spark/blob/46332d985e52f76887c139f67d1471b82e85d5ca/sql/catalyst/src/main/scala/org/apache/spark/sql/types/Decimal.scala#L495-L496).
Because it not bind with round up. So if we change `MathContext` in `Decimal.scala`. We also should round up in  `Decimal.scala` again to change PRECISION to 38. So we will have three times round up. It will not fix the bug.